restli-tools/src/main/java/com/linkedin/restli/tools/data/ExtensionSchemaValidationCmdLineApp.java

```diff
@@ -132,7 +139,29 @@ private static void parseAndValidateExtensionSchemas(String resolverPath, File i
           .stream()
           .filter(f -> !((RecordDataSchema) topLevelDataSchema).isFieldFromIncludes(f))
           .collect(Collectors.toList());
-      // TODO : Call annotation process to validate each field's annotation, need to create schemaVisitor and handler for extension annotation.
+
+      for (RecordDataSchema.Field field : extensionSchemaFields) {
+        Map<String, Object> properties = field.getProperties();
+
+        if (properties.isEmpty() || properties.keySet().size() != 1 || !properties.containsKey(
```

Comment:
I feel manual test for these is not enough. Can you find a way to write test cases for this line of logic?

Reply:
Adding test cases is not that straight forward. There are multiple validations in this method. If we only want to test this specific logic, it is hard to generate extensionSchemaFields. Why do you think manual test is not enough? I tested following scenarios: 
1. extension schema field with @extension
2. extension schema field without any annotation
3. extension schema field with annotation but not with "extension" namespace
4. multiple extension schema fields, some of them with validate annotation, some of them without validate annotation.